moveit_core/robot_model/src/planar_joint_model.cpp

```diff
@@ -127,7 +134,11 @@ void moveit::core::PlanarJointModel::getVariableRandomPositionsNearBy(random_num
     values[1] = rng.uniformReal(std::max(bounds[1].min_position_, near[1] - distance),
                                 std::min(bounds[1].max_position_, near[1] + distance));
 
-  double da = angular_distance_weight_ * distance;
+  double da = 0.0;
+  if (fabs(angular_distance_weight_) > std::numeric_limits<double>::epsilon())
+    da = distance / angular_distance_weight_;
+  if (da > boost::math::constants::pi<double>())
+    da = boost::math::constants::pi<double>();
```

Comment:
I disagree with this change.
The rest of the code reads like `angular_distance_weight_` should always be `0 <= angular_distance_weight_ <= 1`
and afaics from reading the code this should work fine.
Did you find code that disagrees with this assumption in MoveIt?
At the moment I believe it would make more sense to add a safe-guard to `setAngularDistanceWeight` to restrict the value to that range instead.


I agree in that `da` should be bounded to `da <= π`.